### PR TITLE
Ts all the tests

### PR DESCRIPTION
**Pull request by balloob**
Convert our tests to TypeScript to unblock #680

### CODE REVIEW

File: src/common/datetime/seconds_to_duration.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-const leftPad = (number) => (number < 10 ? `0${number}` : number);
+const leftPad = (num) => (num < 10 ? `0${num}` : num);
```
Comment (iantrich) [inline src/common/datetime/seconds_to_duration.ts]: types?

File: src/common/entity/binary_sensor_icon.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /** Return an icon representing a binary sensor state. */
 
 export default function binarySensorIcon(state) {
```
Comment (iantrich) [inline src/common/entity/binary_sensor_icon.ts]: `state: HassEntity`

File: src/common/entity/cover_icon.ts [hunk 3]
```diff
@@ -1,8 +1,9 @@
 /** Return an icon representing a cover state. */
+import { HassEntity } from "home-assistant-js-websocket";
 import domainIcon from "./domain_icon";
 
-export default function coverIcon(state) {
-  var open = state.state && state.state !== "closed";
+export default function coverIcon(state: HassEntity) {
```
Comment (iantrich) [inline src/common/entity/cover_icon.ts]: return type `: string`?

File: src/common/entity/domain_icon.ts [hunk 4]
```diff
@@ -44,7 +44,7 @@ const fixedIcons = {
   weblink: "hass:open-in-new",
 };
 
-export default function domainIcon(domain, state) {
+export default function domainIcon(domain: string, state?: string) {
```
Comment (iantrich) [inline src/common/entity/domain_icon.ts]: return type `: string`

File: src/common/entity/extract_views.ts [hunk 5]
```diff
@@ -1,8 +1,9 @@
+import { HassEntities, HassEntity } from "home-assistant-js-websocket";
 import { DEFAULT_VIEW_ENTITY_ID } from "../const";
 
 // Return an ordered array of available views
-export default function extractViews(entities) {
-  const views = [];
+export default function extractViews(entities: HassEntities) {
```
Comment (iantrich) [inline src/common/entity/extract_views.ts]: return type `: HassEntity[]`

File: test-mocha/common/entity/compute_state_display.ts [hunk 6]
```diff
@@ -3,10 +3,9 @@ import { assert } from "chai";
 import computeStateDisplay from "../../../src/common/entity/compute_state_display";
 
 describe("computeStateDisplay", () => {
-  const localize = function(message, ...args) {
-    // Mock Localize function for testing
-    return message + (args.length ? ": " + args.join(",") : "");
-  };
+  // Mock Localize function for testing
+  const localize = (message, ...args) =>
```
Comment (iantrich) [inline test-mocha/common/entity/compute_state_display.ts]: `message: string`?

Comment (iantrich) [inline test-mocha/common/entity/compute_state_display.ts]: `message: string`

File: test-mocha/common/entity/compute_state_display.ts [hunk 7]
```diff
@@ -35,9 +34,10 @@ describe("computeStateDisplay", () => {
   });
 
   it("Localizes binary sensor invalid device class", () => {
-    const altLocalize = function(message, ...args) {
-      if (message === "state.binary_sensor.invalid_device_class.off")
+    const altLocalize = (message, ...args) => {
```
Comment (iantrich) [inline test-mocha/common/entity/compute_state_display.ts]: `message: string`

File: test-mocha/common/entity/compute_state_display.ts [hunk 8]
```diff
@@ -65,8 +65,10 @@ describe("computeStateDisplay", () => {
   });
 
   it("Localizes unknown sensor value with units", () => {
-    const altLocalize = function(message, ...args) {
-      if (message === "state.sensor.unknown") return null;
+    const altLocalize = (message, ...args) => {
```
Comment (iantrich) [inline test-mocha/common/entity/compute_state_display.ts]: `message: string`

File: test-mocha/common/entity/compute_state_display.ts [hunk 9]
```diff
@@ -83,8 +85,10 @@ describe("computeStateDisplay", () => {
   });
 
   it("Localizes unavailable sensor value with units", () => {
-    const altLocalize = function(message, ...args) {
-      if (message === "state.sensor.unavailable") return null;
+    const altLocalize = (message, ...args) => {
```
Comment (iantrich) [inline test-mocha/common/entity/compute_state_display.ts]: `message: string`

File: test-mocha/common/entity/compute_state_display.ts [hunk 10]
```diff
@@ -101,8 +105,10 @@ describe("computeStateDisplay", () => {
   });
 
   it("Localizes sensor value with component translation", () => {
-    const altLocalize = function(message, ...args) {
-      if (message !== "component.sensor.state.custom_state") return null;
+    const altLocalize = (message, ...args) => {
```
Comment (iantrich) [inline test-mocha/common/entity/compute_state_display.ts]: `message: string`

File: test-mocha/common/entity/compute_state_display.ts [hunk 11]
```diff
@@ -220,8 +226,10 @@ describe("computeStateDisplay", () => {
   });
 
   it("Localizes unavailable", () => {
-    const altLocalize = function(message, ...args) {
-      if (message === "state.sensor.unavailable") return null;
+    const altLocalize = (message, ...args) => {
```
Comment (iantrich) [inline test-mocha/common/entity/compute_state_display.ts]: `message: string`

**Comment by iantrich** [comment]
Looking over other people's changes is hard! 😃

**Comment by balloob** [comment]
You're right Ian, I went lazy and didn't add types to all methods, guess I converted too many 🤷‍♂️  Anyway, new day, new changes and just spend some time polishing this PR and adding the types.

**Comment by balloob** [comment]
For the types in the test file: they are really just for tests and don't need to have extensive typing. I've used a bunch of `any` to get it to pass.

**Pull request merged**